src/ssort/_graphs.py

```diff
@@ -97,8 +103,10 @@ def _find_cycle(graph):
                 stack.append((dependency, set(graph.dependencies[dependency])))
                 in_stack.add(dependency)
 
+    return None
+
 
-def replace_cycles(graph, *, key):
+def replace_cycles(graph: Graph[_T], *, key: Callable[[_T], Any]) -> None:
```

Comment:
I think `key` is meant to be `Callable[[_T], int]`.  I think my comment further down, that this is an awkward way to indicate ordering, still stands.

Reply:
I see. I just attempted to mirror the type annotation of `key` parameter from the `sorted` function, which I got from [here](https://github.com/python/typeshed/blob/master/stdlib/builtins.pyi#L1468). Of course, that annotation, `Callable[[_T], SupportsRichComparison]`,  includes `SupportsRichComparison` which is not part of the standard library so I replaced it with `Any`.

Regardless, I don't see any issues with narrowing the return type of `key` from `Any` to `int`.